eZ/Publish/Core/Persistence/Legacy/UserPreference/Mapper.php

```diff
@@ -42,7 +42,7 @@ private function extractUserPreferenceFromRow(array $row): UserPreference
         $userPreference->id = (int)$row['id'];
         $userPreference->userId = (int)$row['user_id'];
         $userPreference->name = $row['name'];
-        $userPreference->value = $row['value'];
+        $userPreference->value = strip_tags($row['value']);
```

Comment:
The PHP doc for strip_tags says _"Warning: This function should not be used to try to prevent XSS attacks. Use more appropriate functions like htmlspecialchars() or other means depending on the context of the output."_

Have you considered htmlspecialchars(), and if so, why did you go for strip_tags?

Reply:
@glye thank you for your suggestion. The whitelisting is also implemented and fixed in the second part of CR: https://github.com/ezsystems/ezplatform-admin-ui/pull/1211/files . This strip_tags is only for read from database and looks sufficient in our context, but there is nothing against to use htmlspecialchars()

Reply:
I see, that's great. Given the whitelisting the strip_tags doesn't really matter, but at least we can avoid some false positives by not using it for this kind of thing. 👍